gotocompany/compass/v1beta1/service.proto

```diff
@@ -816,6 +832,52 @@ message SearchAssetsRequest {
   }];
 }
 
+
+message GroupAssetsRequest {
+
+  string groupby = 1 [
+    (validate.rules).string = {
+      ignore_empty: false,
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "fields used to group assets. multiple fields can also be provided here"
+    }
+  ];
+
+  map<string, string> filter = 2 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+    description: "filter result based on a (nested) field of the asset. the nested field is written with period separated field name. eg, \"filter[data.landscape]\""
+  }];
+
+  string include_fields = 3 [(validate.rules).string = {
+    ignore_empty: true
+  }];
+
+  uint32 size = 4 [
+    (validate.rules).uint32 = {
+      gte: 0,
+      ignore_empty: true
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "maximum size to fetch"
+    }
+  ];
+
+}
+
+message GroupAssetsResponse {
+  repeated GroupAssetInfo group_asset_info = 1;
+}
+
+message GroupAssetInfo {
+  repeated GroupFieldInfo group_field_info = 1;
+  repeated Asset data = 2;
+}
+
+message GroupFieldInfo {
```

Comment:
If we can group on multiple fields, why is there a single grouping key and value? Should it be a repeated list of key and value instead?

Reply:
It is repeated under GroupAssetsInfo. Updated the sample here: https://docs.google.com/document/d/18qBCWhu_OaHwTKl_URsGJQoqQx8mVulDlFxVWAHO5P8/edit#